### PR TITLE
[ADVAPP-127] & [ADVAPP-128]: Creating and Managing SLAs

### PR DESCRIPTION
**Pull request by danharrin**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/ADVAPP-127
- https://canyongbs.atlassian.net/browse/ADVAPP-128

### Technical Description

Implemented SLAs, related to a type-priority. Monitored compliance with Eloquent rather than on the database level for now.

### Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Content or styling update (Changes which don't affect functionality)
- [ ] DevOps
- [ ] Documentation

### Screenshots (if appropriate)

https://github.com/canyongbs/advisingapp/assets/41773797/b5003ae6-67db-462a-b7c0-524493fe3d61

### Any deployment steps required?

A deployment step could be a command that needs to be executed or an ENV key that needs to be added, for example.

- [ ] Yes, please specify
- [x] No

_______________________________________________

## Before contributing and submitting this PR, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/canyongbs/advisingapp/blob/main/README.md#contributing).
* [x] Title the PR with the ticket/issue number and a short description of the changes made. Or if no ticket/issue exists, title the PR with a short description of the changes made
* [x] Linked a relevant ticket or issue or describe the issue/feature which this PR resolves/implements.
* [x] Resolved all conflicts, if any.
* [x] Rebased your branch PR on top of the latest upstream `develop` branch.

### CODE REVIEW

File: app-modules/service-management/config/roles/web/slat_management.php
Comment (Orrison) [inline app-modules/service-management/config/roles/web/slat_management.php:1]: Is there a typo here with `slat_management`?

File: app-modules/service-management/database/migrations/2023_09_05_143920_create_slas_table.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        });
+    }
+
+    /**
+     * Reverse the migrations.
+     */
+    public function down(): void
+    {
+        Schema::dropIfExists('slas');
+    }
```
Comment (Orrison) [inline app-modules/service-management/database/migrations/2023_09_05_143920_create_slas_table.php:65]: Let's not include the `down` here. None of our other migrations do or should

File: app-modules/service-management/src/Filament/Resources/SlaResource.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            ServiceRequestPrioritiesRelationManager::class,
+        ];
+    }
+
+    public static function getPages(): array
+    {
+        return [
+            'index' => Pages\ListSlas::route('/'),
+            'create' => Pages\CreateSla::route('/create'),
+            'edit' => Pages\EditSla::route('/{record}/edit'),
```
Comment (Orrison) [inline app-modules/service-management/src/Filament/Resources/SlaResource.php:96]: Let's complete the use statements here.

File: app-modules/service-management/src/Filament/Resources/SlaResource/Pages/ListSlas.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    public function table(Table $table): Table
+    {
+        return $table
+            ->columns([
+                TextColumn::make('name')
+                    ->searchable(),
+            ])
+            ->filters([
+            ])
```
Comment (Orrison) [inline app-modules/service-management/src/Filament/Resources/SlaResource/Pages/ListSlas.php:60]: Let's remove these empty methods if we are not going to use them for now.

**Review by Orrison — CHANGES_REQUESTED** [review]

File: app-modules/service-management/config/roles/web/slat_management.php
Comment (danharrin) [reply app-modules/service-management/config/roles/web/slat_management.php:1]: Fixed

File: app-modules/service-management/database/migrations/2023_09_05_143920_create_slas_table.php (hunk 1 above)
Comment (danharrin) [reply app-modules/service-management/database/migrations/2023_09_05_143920_create_slas_table.php:65]: ok, updated the stubs

File: app-modules/service-management/src/Filament/Resources/SlaResource.php (hunk 2 above)
Comment (danharrin) [reply app-modules/service-management/src/Filament/Resources/SlaResource.php:96]: ok, also fixed elsewhere

File: app-modules/service-management/src/Filament/Resources/SlaResource/Pages/ListSlas.php (hunk 3 above)
Comment (danharrin) [reply app-modules/service-management/src/Filament/Resources/SlaResource/Pages/ListSlas.php]: ok, also done elsewhere

**Review by Orrison — APPROVED** [review]

**Pull request merged**